### PR TITLE
added initial db implementation

### PR DESCRIPTION
**Pull request by mikitachab**
This PR is solution of #5 
- added initials ORM models implementation
- now it uses sqlite driver
- added DBtest function which is basically kind of a showcase how to use this db ORM

### CODE REVIEW

File: db/db.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"time"
+
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+)
+
+func GetDB() *gorm.DB {
+	db, err := gorm.Open(sqlite.Open("test.db"), &gorm.Config{})
+	if err != nil {
+		panic("failed to connect database")
```
Comment (antonskwr) [inline db/db.go]: Do we use panic or log.Fatal? To have some information about the time the failure occurred maybe? We should add to the spec the way we handle errors.

File: db/db.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		Player: player1, SevenWondersScore: score})
+	db.Create(&PlayScore{Play: play,
+		Player: player2, SevenWondersScore: score2})
+
+	var playScores []PlayScore
+
+	db.Joins("Player").Joins("Play").Joins("SevenWondersScore").Find(&playScores)
+
+	for _, ps := range playScores {
+		fmt.Println(ps.Player.Name, ps.SevenWondersScore.ScoreSum())
```
Comment (antonskwr) [inline db/db.go]: Maybe formatted print might be a little more informative?

File: db/models.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		sws.MilitaryConflicts +
+		sws.Wonders +
+		sws.TreasuryContents +
+		sws.CommercialStructures +
+		sws.Guilds
+}
+
+type PlayScore struct {
+	gorm.Model
+	PlayerID            uint
```
Comment (antonskwr) [inline db/models.go:51]: Should we go with uint or int? As sqlite uses signed integers, maybe we should use int for IDs to be coherent with the database.

File: db/models.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	TreasuryContents     int
+	CivilianStructures   int
+	ScientificStructures int
+	CommercialStructures int
+	Guilds               int
+
+	PlayScores []PlayScore
+}
+
+func (sws *SevenWondersScore) ScoreSum() int {
```
Comment (antonskwr) [inline db/models.go]: GetScoreSum or CalculateScoreSum, since "score" is a verb, might be a little confusing later on.

File: db/db.go (hunk 1 above)
Comment (mikitachab) [reply db/db.go]: yeah, a will change it to `log.Fatal`. And I agree that we should do kind of coding guidelines spec

File: db/db.go (hunk 2 above)
Comment (mikitachab) [reply db/db.go]: this is just a starter code that meant to be removed, so I'm not sure that we should improve it

File: db/models.go (hunk 3 above)
Comment (mikitachab) [reply db/models.go:51]: > coherent with the database

whole point using ORM is to do not care what happening at DB level
>sqlite uses signed integers

and do not stick to specific database. sqlite just for dev. We want to use PostgreSQL in prod.

File: db/models.go (hunk 4 above)
Comment (mikitachab) [reply db/models.go]: yeah, I should change it

File: db/db.go (hunk 1 above)
Comment (mikitachab) [reply db/db.go]: also I want to notice that db setup happening once at startup

Comment (mikitachab) [reply db/db.go]: >  change it to log.Fatal

done

File: db/models.go (hunk 4 above)
Comment (mikitachab) [reply db/models.go]: done

**Review by antonskwr — APPROVED** [review]

**Pull request merged**